packages/terra-site/src/ThemeProvider.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /**
+   * The component(s) that will be wrapped by <ThemeProvider />
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Name of class for specified theme. e.g `terra-OSC-theme`
+   */
+  themeName: PropTypes.string.isRequired,
+};
+
+const ThemeProvider = ({ themeName, children, ...customProps }) => {
+  return <div className={themeName}>{children}</div>;
```

Comment:
After this PR lands, I plan to have `<Base />` wrap the `<ThemeProvider />` component so people can interact with it through Base via `<Base themeName="my-theme">`.

`<ThemeProvider />` could still be nested within `<Base themeName="my-theme">` if people wanted nested theming e.g. 
```js
<Base themeName="my-theme">
  <ThemeProvider themeName="dark">
    <Sidebar content={<SidebarContent/>} />
  </ThemeProvider>
  <Body content={<BodyContent/>} />
</Base>